src/world/World.cpp

```diff
@@ -8,7 +8,10 @@ namespace rtt::ai::world {
 World worldObj;
 World *world = &worldObj;
 
-void World::updateWorld(const Field &field, const proto::World &message) {
+void World::updateWorld(const proto::SSL_GeometryFieldSize &fieldMessage, const proto::World &worldMessage) {
+    ai::Field field(fieldMessage);
+    this->field = field;
```

Comment:
Does this work? What happens to "this->field" if "field" goes out of scope?

Reply:
nothing it is an implicit move